src/api/CheckListItemApi.ts

```diff
@@ -163,6 +231,48 @@ const getCheckListItemsByRequestId = async (RequestId: number) => {
   }
 };
 
+/** Internal functions that actually do the fetching
+ * @returns The ICheckListItems that are Open (do not have a completeion date) -- TODO -- Cancelled??
+ */
+const getOpenCheckListItems = async () => {
+  if (process.env.NODE_ENV === "development") {
+    await sleep(2000);
+    return Promise.resolve(
+      testCheckListItems.filter((item) => !item.CompletedDate)
+    );
+  } else {
+    try {
+      return spWebContext.web.lists
+        .getByTitle("CheckListItems")
+        .items.filter("CompletedDate eq null")
+        .select(requestedFields)
+        .expand(expandedFields)();
+    } catch (e) {
+      console.error(
+        `Error occurred while trying to fetch Open CheckListeItems`
+      );
+      console.error(e);
+      if (e instanceof Error) {
+        throw new ApiError(
+          e,
+          `Error occurred while trying to fetch Open CheckListeItems: ${e.message}`
+        );
+      } else if (typeof e === "string") {
+        throw new ApiError(
+          new Error(
+            `Error occurred while trying to fetch Open CheckListeItems: ${e}`
+          )
+        );
+      } else {
+        throw new ApiError(
+          undefined,
+          `Unknown error occurred while trying to fetch Open CheckListeItems`
+        );
+      }
+    }
```

Comment:
Yes, agree.  But suggest we open a separate issue for this.  I started messing with it a little in this PR by adding a way for the update Checklist to pass back the error to the caller, because it is exactly like you said, in that the try/catch throws the error but nothing catches it.

Reply:
Removed this try/catch and moved logic into the onError along with the logic from MyCheckListItem for the update to create the notification.